### PR TITLE
VACMS-2572: Add patch and empty checks to prevent empty array notices for tablefield.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #2572 

## Testing done
Drupal log review

## Screenshots
N/A

## QA steps

- Open drupal logs, and filter by php errors. Take note of time of most recent entry for `Notice: Undefined index: caption in Drupal\tablefield\Plugin\Field\FieldFormatter\TablefieldFormatter->viewElements() (line 145 of /app/docroot/modules/contrib/tablefield/src/Plugin/Field/FieldFormatter/TablefieldFormatter.php)`
- Go to `/va-oklahoma-city-health-care/locations/ada-va-clinic` and refresh page
- Go back to log page, filter by php, and confirm there aren't any newer instances of `Notice: Undefined index: caption in Drupal\tablefield\Plugin\Field\FieldFormatter\TablefieldFormatter->viewElements() (line 145 of /app/docroot/modules/contrib/tablefield/src/Plugin/Field/FieldFormatter/TablefieldFormatter.php)`

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -2935,26 +2935,26 @@
         },
         {
             "name": "drupal/auto_entitylabel",
-            "version": "3.0.0-beta3",
+            "version": "2.1.0-beta1",
             "source": {
                 "type": "git",
                 "url": "https://git.drupalcode.org/project/auto_entitylabel.git",
-                "reference": "8.x-3.0-beta3"
+                "reference": "8.x-2.1-beta1"
```
Comment (swirtSJW) [inline composer.lock:2942]: We're rolling back to an older version?

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -437,7 +437,9 @@ function _va_gov_backend_modify_paragraph_interactions(array &$element, FormStat
       ];
     }
     // Remove catpion, rows, and import fields from address tablefield.
-    unset($element['subform']['field_facility_service_hours']['widget'][0]['caption']);
+    if (!empty($element['subform']['field_facility_service_hours']['widget'][0]['caption'])) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:440]: Moderate:  Should this be unset if it is empty?  Right now it is only going to unset it if it is filled.   I think this should be checking only isset().
```suggestion
    if (isset($element['subform']['field_facility_service_hours']['widget'][0]['caption'])) {
```

File: docroot/modules/custom/va_gov_consumers/va_gov_consumers.module [hunk 3]
```diff
@@ -79,7 +79,9 @@ function _vagov_consumers_disable_vha_api_form_fields(array &$form, $form_id) {
     // Disable items that are populated by the Facility API migration.
     $form['title']['widget'][0]['value']['#attributes']['disabled'] = TRUE;
     $form['field_facility_classification']['widget']['#attributes']['disabled'] = TRUE;
-    unset($form['field_facility_hours']['widget'][0]['caption']);
+    if (!empty($form['field_facility_hours']['widget'][0]['caption'])) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_consumers/va_gov_consumers.module:82]:
```suggestion
    if (isset($form['field_facility_hours']['widget'][0]['caption'])) {
```

**Comment by ethanteague** [comment]
> Double checking the logic. Made suggestions based on the idea that if this exists, it should always be unset. If my thinking is bad, push back ;)

Makes sense to me, @swirtSJW :+1: committed.

**Comment by swirtSJW** [comment]
Looks like tests failed for some unrelated reason 
image


I just re-triggered tests.

**Comment by swirtSJW** [comment]
Something seems odd on this one @ethanteague.  Your !empty checks came back and now we seem to a have a bunch of behat stuff that seems unrelated.

**Pull request merged**